pages/src/utils/childDocumentsOf.ts

```diff
@@ -0,0 +1,77 @@
+import { PayloadRequest, CollectionSlug } from 'payload'
+import type { SanitizedCollectionConfig } from 'payload'
+import type { PageCollectionConfigAttributes } from '../types/PageCollectionConfigAttributes.js'
+import type { PagesPluginConfig } from '../types/PagesPluginConfig.js'
+
+async function childDocumentsOfInternal(
+  req: PayloadRequest,
+  docId: string | number,
+  collectionSlug: CollectionSlug,
+  baseFilter?: PagesPluginConfig['baseFilter']
+): Promise<{ id: string | number; collection: CollectionSlug }[]> {
+  const childReferences: { id: string | number; collection: CollectionSlug }[] = []
+  
+  const allCollections = req.payload.config.collections || []
+  
+  const pageCollections = allCollections.filter(
+    isPageCollectionWithParent(collectionSlug)
+  )
+  
+  for (const targetCollection of pageCollections) {
+    const parentFieldName = targetCollection.page.parent.name || 'parent'
+    
+    const baseFilterWhere = typeof baseFilter === 'function' 
+      ? baseFilter({ req }) 
+      : undefined
+    
+    try {
+      const childDocuments = await req.payload.find({
+        collection: targetCollection.slug,
+        where: {
+          and: [
+            { [parentFieldName]: { equals: docId } },
+            ...(baseFilterWhere ? [baseFilterWhere] : []),
+          ],
+        },
+        depth: 0,
+        select: {},
+        limit: 0,
+      })
+      
+      for (const doc of childDocuments.docs) {
+        childReferences.push({
+          id: doc.id,
+          collection: targetCollection.slug,
+        })
+      }
+    } catch (error) {
+      console.warn(`Error checking collection ${targetCollection.slug} for child documents:`, error)
+    }
+  }
+  
+  return childReferences
+}
+
+export const childDocumentsOf = childDocumentsOfInternal
+
+export async function hasChildDocuments(
```

Comment:
All exported functions should have a short doc comment. Please add one here to since we export this function in index.ts